app/test/consistent_apphash_test.go

```diff
@@ -171,3 +340,144 @@ func deterministicKeyRing(cdc codec.Codec) (keyring.Keyring, []types.PubKey) {
 	}
 	return kb, pubKeys
 }
+
+// processSdkMessages takes a list of sdk messages, forms transactions, signs them
+// and returns a list of raw transactions
+func processSdkMessages(signer *user.Signer, sdkMessages []sdk.Msg) ([][]byte, error) {
+	rawSdkTxs := make([][]byte, 0, len(sdkMessages))
+	for _, msg := range sdkMessages {
+		rawSdkTx, err := signer.CreateTx([]sdk.Msg{msg}, blobfactory.DefaultTxOpts()...)
+		if err != nil {
+			return nil, err
+		}
+
+		signerAddress := msg.GetSigners()[0]
+		signerAccount := signer.AccountByAddress(signerAddress)
+		err = signer.SetSequence(signerAccount.Name(), signerAccount.Sequence()+1)
+		if err != nil {
+			return nil, err
+		}
+
+		rawSdkTxs = append(rawSdkTxs, rawSdkTx)
+	}
+	return rawSdkTxs, nil
+}
+
+// executeTxs executes a set of transactions and returns the data hash and app hash
+func executeTxs(testApp *app.App, rawBlobTx []byte, rawSdkTxs [][]byte, validators []abci.Validator, lastCommitHash []byte) ([]byte, []byte, error) {
```

Comment:
[non-blocking]

imo, some version of this would be great to add directly to testapp to use elsewhere!